pandas/core/indexes/base.py

```diff
@@ -3848,7 +3848,16 @@ def join(self, other, how="left", level=None, return_indexers=False, sort=False)
                 return self._join_non_unique(
                     other, how=how, return_indexers=return_indexers
                 )
-        elif self.is_monotonic and other.is_monotonic:
+        elif (
+            self.is_monotonic
+            and other.is_monotonic
+            and (
+                not isinstance(self, ABCMultiIndex)
```

Comment:
can you check _is_lexsorted instead rather than this specific check.

Reply:
Same issue as with is_monotonic, depends on the categories too.